### PR TITLE
Extend check controller with server side apply

### PR DESCRIPTION
**Pull request by pperiyasamy**
This extends connectivity check controller's sync function with server side apply so that check objects can be applied directly on the K8s without need for checking existing object fields and calling update function.

Signed-off-by: Periyasamy Palanisamy <pepalani@redhat.com>

### CODE REVIEW

**Comment by pperiyasamy** [comment]
> @deads2k @tkashem @dgrisonnet

This is tested with required changes in connectivity check controller in cno https://github.com/openshift/cluster-network-operator/pull/1649, it seems to be working perfectly fine. I can also verify conflict and ownership transfer cases with `kubectl apply` and `kubectl edit `operations.

File: pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go [hunk 1]
```diff
@@ -186,11 +196,39 @@ func (c *connectivityCheckController) Sync(ctx context.Context, syncContext fact
 		}
 	}
 
-	newChecks, err := c.podNetworkConnectivityCheckFn(ctx, syncContext)
+	var newCheckNames sets.String
+	if c.podNetworkConnectivityCheckFn != nil {
+		newCheckNames, err = c.handlePodNetworkConnectivityCheckFn(ctx, syncContext)
```
Comment (deads2k) [inline pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go:201]: any plan to retire this function once the transition is made?  I prefer to have a single path if possible.

**Comment by deads2k** [comment]
At some point a lister may be necessary to reduce QPS, but this is a significant improvement over the current logic

/lgtm

**Pull request merged**